packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -82,6 +88,22 @@ abstract class WidgetsBinding extends BindingBase implements GestureBinding, Ren
         buildOwner.reassemble(renderViewElement);
       }
     );
+
+    registerStringServiceExtension(
+      name: 'applicationRoute',
+      getter: () => _navigatorState?.currentRoute?.settings?.name,
+      setter: (String route) {
+        _navigatorState?.pushNamed(route);
+      }
+    );
+  }
+
+  NavigatorState get _navigatorState {
+    NavigatorOwner observer = WidgetsBinding.instance._observers.firstWhere(
```

Comment:
I suspect there's a better way to structure this interaction.  Having the binding reach down in the object hierarchy to magically discover that the WidgetsBindingObserver is also a NavigatorOwner is pretty gross.